web/packages/teleport/src/components/FormLogin/FormIdentifierFirst.tsx

```diff
@@ -0,0 +1,275 @@
+/**
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { useCallback, useEffect, useState } from 'react';
+
+import {
+  Box,
+  ButtonLink,
+  ButtonPrimary,
+  Flex,
+  Indicator,
+  Input,
+  Text,
+} from 'design';
+import * as Alerts from 'design/Alert';
+import ButtonSso, { guessProviderType } from 'shared/components/ButtonSso';
+import { useRefAutoFocus } from 'shared/hooks';
+import { Attempt, useAsync } from 'shared/hooks/useAsync';
+import { AuthProvider } from 'shared/services';
+
+import ResourceService from 'teleport/services/resources';
+import { storageService } from 'teleport/services/storageService';
+
+import SSOButtonList from './SsoButtons';
+
+type Props = {
+  onLoginWithSso(provider: AuthProvider): void;
+  /**
+   * onUseLocalLogin is called to switch the view to the local login form.
+   */
+  onUseLocalLogin(): void;
+};
+
+export function FormIdentifierFirst({
+  onLoginWithSso,
+  onUseLocalLogin,
+}: Props) {
+  const [resourceService] = useState(() => new ResourceService());
+
+  const [rememberedUsername, setRememberedUsername] = useState<string>(
+    storageService.getRememberedSsoUsername()
+  );
+  const [username, setUsername] = useState<string>(rememberedUsername);
+  const [connectors, setConnectors] = useState<AuthProvider[]>([]);
+
+  useEffect(() => {
+    if (rememberedUsername) {
+      fetchMatchingConnectors(rememberedUsername);
+    }
+  }, [rememberedUsername]);
+
+  const [fetchAttempt, fetchMatchingConnectors] = useAsync(
+    useCallback(
+      async (username: string) => {
+        const connectors =
+          await resourceService.getUserMatchedAuthConnectors(username);
```

Comment:
Reading https://github.com/gravitational/teleport/blob/master/rfd/0214-identifier-first-login.md#implementation, if the identifier login is enabled (which we learn from response to `config.js`, we will hit the `/authconnectors` again? 
For situation where the user is already remembered in the frontend, can't we just get the matched connector in response to `/config.js`?  Or this is the case already?

Reply:
No, since the request for `web/config.js` isn't a typical request that we make in the frontend and it's actually made before we even render the UI, so it's not straightforward to retrieve the saved user and then include it as a query parameter.

Reply:
Gotcha. In that case, should we stop quering connectors in the `web/config.js` request and always fetch them with the new `authconnectors` endpoint? 

Not necessarily in this PR but it seems like something to consider

Reply:
We need to do it here just to know whether or not identifier first login is enabled so that we know whether to render the identifier first login forms in the UI, and the only way to check if it's enabled is by querying connectors and seeing if they have `user_matchers` defined.

Reply:
Essentially the flow is

User loads the Web UI -> WebUI fetches config before rendering -> If the config has `identifierFirstLoginEnabled: true`, WebUI renders the identifier first login form.